anomalib/data/mvtec_3d.py

```diff
@@ -0,0 +1,293 @@
+    samples.loc[(samples.label != "good"), "label_index"] = 1
+    samples.label_index = samples.label_index.astype(int)
+
+    # separate masks from samples
+    mask_samples = samples.loc[((samples.split == "test") & (samples.type == "rgb"))].sort_values(
+        by="image_path", ignore_index=True
+    )
+    samples = samples.sort_values(by="image_path", ignore_index=True)
+
+    # assign mask paths to all test images
+    samples.loc[((samples.split == "test") & (samples.type == "rgb")), "mask_path"] = (
+        mask_samples.path + "/" + samples.split + "/" + samples.label + "/" + "gt/" + samples.file_name
+    )
+    samples.dropna(subset=["image_path"], inplace=True)
+    samples = samples.astype({"image_path": "str", "mask_path": "str", "depth_path": "str"})
+
+    # assert that the right mask files are associated with the right test images
+    assert (
+        samples.loc[samples.label_index == 1]
+        .apply(lambda x: Path(x.image_path).stem in Path(x.mask_path).stem, axis=1)
+        .all()
+    ), "Mismatch between anomalous images and ground truth masks. Make sure the mask files in 'ground_truth' \
+              folder follow the same naming convention as the anomalous images in the dataset (e.g. image: '000.png', \
+              mask: '000.png' or '000_mask.png')."
+
+    # assert that the right depth image files are associated with the right test images
+    assert (
+        samples.loc[samples.label_index == 1]
+        .apply(lambda x: Path(x.image_path).stem in Path(x.depth_path).stem, axis=1)
+        .all()
+    ), "Mismatch between anomalous images and depth images. Make sure the mask files in 'xyz' \
+              folder follow the same naming convention as the anomalous images in the dataset (e.g. image: '000.png', \
+              depth: '000.tiff')."
+
+    if split:
+        samples = samples[samples.split == split].reset_index(drop=True)
+
+    return samples
+
+
+class MVTec3DDataset(AnomalibDepthDataset):
+    """MVTec 3D dataset class.
+
+    Args:
+        task (TaskType): Task type, ``classification``, ``detection`` or ``segmentation``
+        transform (A.Compose): Albumentations Compose object describing the transforms that are applied to the inputs.
+        split (str | Split | None): Split of the dataset, usually Split.TRAIN or Split.TEST
+        root (Path | str): Path to the root of the dataset
+        category (str): Sub-category of the dataset, e.g. 'bagel'
+    """
+
+    def __init__(
+        self,
+        task: TaskType,
+        transform: A.Compose,
+        root: Path | str,
+        category: str,
+        split: str | Split | None = None,
+    ) -> None:
+        super().__init__(task=task, transform=transform)
+
+        self.root_category = Path(root) / Path(category)
+        self.split = split
+
+    def _setup(self) -> None:
+        self.samples = make_mvtec_3d_dataset(self.root_category, split=self.split, extensions=IMG_EXTENSIONS)
+
+
+class MVTec3D(AnomalibDataModule):
```

Comment:
Can you add an entry for this module in `anomalib.data.get_datamodule` so that the dataset can be used from the `config.yaml`?